include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -631,6 +631,12 @@ namespace VectorTools
         if (!cell1->active() && !cell2->active())
           continue;
 
+        // skip foreign cells
+        if (cell1->active() && !cell1->is_locally_owned())
+          continue;
+        if (cell2->active() && !cell2->is_locally_owned())
+          continue;
```

Comment:
It is non-trivial to check this because one of the meshes can be more refined. Right now you get an assertion in ```get_dof_values()``` or ```set_dof_values_by_interpolation()```. I can add a check in debug mode, though.